dask/async.py

```diff
@@ -458,16 +463,28 @@ def fire_task():
             data = dict((dep, state['cache'][dep])
                         for dep in get_dependencies(dsk, key))
             # Submit
-            apply_async(execute_task, args=[key, dsk[key], data, queue,
-                                            get_id, raise_on_exception])
+            apply_async(execute_task,
+                        args=(key, marshall((dsk[key], data)),
+                              marshall, unmarshall,
+                              get_id, raise_on_exception),
+                        callback=queue.put)
+
+        def fill_pool_with_ready_tasks():
+            while state['ready'] and len(state['running']) < num_workers:
+                fire_task()
```

Comment:
slight preference to keep the loop out of a function, even though the code is repeated.